k8s/kubevirt-dynamic/virtualmachineinstancemigration.go

```diff
@@ -214,15 +214,24 @@ func (c *Client) unstructuredGetVMIMigration(
 		return nil, fmt.Errorf("failed to get 'phase' from the vmi migration status: %w", err)
 	}
 	// completed
-	ret.Completed, _, err = unstructured.NestedBool(migration.Object, "status", "migrationState", "completed")
+	var found bool
+	ret.Completed, found, err = unstructured.NestedBool(migration.Object, "status", "migrationState", "completed")
 	if err != nil {
 		return nil, fmt.Errorf("failed to get 'completed' from the vmi migration status: %w", err)
 	}
+	// migrationState is not present if the pod fails to get scheduled; we need to look at the Phase
+	if !found && ret.Phase == "Failed" {
+		ret.Completed = true
+	}
 	// failed
-	ret.Failed, _, err = unstructured.NestedBool(migration.Object, "status", "migrationState", "failed")
+	ret.Failed, found, err = unstructured.NestedBool(migration.Object, "status", "migrationState", "failed")
 	if err != nil {
 		return nil, fmt.Errorf("failed to get 'failed' from the vmi migration status: %w", err)
 	}
+	// migrationState is not present if the pod fails to get scheduled; we need to look at the Phase
+	if !found && ret.Phase == "Failed" {
+		ret.Failed = true
```

Comment:
If migration fails and no migrationState is added to status, both the above NestedBool calls will fail. In that case we will end up setting Completed=true but Failed=false(default) which will mean migration completed successfully which is not right.

Reply:
My bad. We are setting Failed to true from the Phase.

Reply:
if migrationState is not present, err is nil and found if false. Additionally, if the phase is "Failed", we will set both ret.Completed and ret.Failed to true. I don't see why Failed will be default in that case.

Reply:
Ok, just saw your followup comment.